### PR TITLE
Handle input types for Connector, JSON / XML schema input

### PR DESCRIPTION
**Pull request by dinuish94**

### CODE REVIEW

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java [hunk 1]
```diff
@@ -1042,14 +1044,37 @@ private void saveInputOutputSchema(String schemaFilePath) {
...
+        if (null != originalFilePath && null != destinationFilePath
+                && (inputType.equals("XML") || inputType.equals("JSON") || inputType.equals("CSV"))) {
+            File originalFile = new File(originalFilePath);
+            File copiedFile = new File(destinationFilePath);
+            try {
+                FileUtils.copyFile(originalFile, copiedFile);
+                assert (copiedFile).exists();
+                assert (Files.readAllLines(originalFile.toPath()).equals(Files.readAllLines(copiedFile.toPath())));
+            } catch (IOException e) {
+                log.error("Failed to create a copy of the original input file!");
```
Comment (mohanvive) [inline plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java]: provide bit more context to the error message.

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java [hunk 2]
```diff
@@ -1042,14 +1044,37 @@ private void saveInputOutputSchema(String schemaFilePath) {
...
-            FileUtils.copyFile(originalFile, copiedFile);
-            assert (copiedFile).exists();
-            assert (Files.readAllLines(originalFile.toPath()).equals(Files.readAllLines(copiedFile.toPath())));
-        } catch (IOException e) {
-            log.error("Failed to create a copy of the original input file!");
+            PrintWriter writer = new PrintWriter(file);
+            writer.print("");
+            writer.close();
+        } catch (FileNotFoundException e) {
+            log.error("Input sample file not found!", e);
```
Comment (mohanvive) [inline plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java]: Add more content to the error message.

Comment (dinuish94) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

File: plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java (hunk 1 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/SchemaKeyEditorDialog.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

**Pull request merged**